### PR TITLE
Fixes install timeouts

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

- Adds a transition to reset BMCs before proceeding with firmware installs.
- Sets a separate firmware install timeout, this fixes the firmware install process timing out on Redfish BMCs. 

related https://github.com/bmc-toolbox/bmclib/pull/337

### CODE REVIEW

File: internal/outofband/actions.go [hunk 1]
```diff
@@ -84,6 +87,11 @@ func actionDocumentation() ([]sw.StateDoc, []sw.TransitionTypeDoc) {
 				Name:        string(stateDownloadedFirmware),
 				Description: "This action state indicates the component firmware to be installed has been downloaded and verified.",
 			},
+			{
+				Name:        string(statePreInstallResetBMC),
+				Description: "This action state indicates the BMC has been power cycled either as a pre-install step to make sure the BMC is in good health before proceeding.",
```
Comment (DoctorVin) [inline internal/outofband/actions.go:92]: Nit: `s/either//`

File: internal/outofband/actions.go [hunk 2]
```diff
@@ -93,8 +101,8 @@ func actionDocumentation() ([]sw.StateDoc, []sw.TransitionTypeDoc) {
 				Description: "This action state indicates the component firmware install status is in a finalized state (powerCycleDevice, powerCycleBMC, successful, failed).",
 			},
 			{
-				Name:        string(stateResetBMC),
-				Description: "This action state indicates the BMC has been (conditionally) power cycled to complete a component firmware install.",
+				Name:        string(statePostInstallResetBMC),
+				Description: "This action state indicates the BMC has been power cycled either as a post-install step to complete a component firmware install.",
```
Comment (DoctorVin) [inline internal/outofband/actions.go:105]: As above, the `either` is extraneous here

**Review by DoctorVin — APPROVED** [review]
Some of the wording is cumbersome, bt the code is fine.

**Comment by joelrebel** [comment]
Thanks for the feedback, I've pushed a fix. Also included are the updated statemachine docs.

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**